src/components/layout/sendTip/GiphySearch.vue

```diff
@@ -0,0 +1,178 @@
+<template>
+  <div class="giphy-search">
+    <div class="search-input">
+      <input
+        v-model="query"
+        type="text"
+        placeholder="Search GIPHY"
+        @input="submit"
+      >
+      <IconSearch />
+    </div>
+    <div
+      v-if="resultsCount"
+      class="results-count"
+    >
+      {{ resultsCount }}
+    </div>
+    <div
+      ref="gifsContainer"
+      class="results"
+      @scroll="loadMore"
+    >
+      <div
+        v-for="(item, colIndex) in Array(3)"
+        :key="colIndex"
+        class="gif-column"
+      >
+        <div
+          v-for="({ still, animated, original }, index) in results"
+          :key="index"
+        >
+          <img
+            v-if="index % 3 === colIndex"
+            :src="still"
+            @mouseenter="toggleStill($event.target, animated)"
+            @mouseleave="toggleStill($event.target, still)"
+            @click="$emit('add-gif', original)"
+          >
+        </div>
+      </div>
+    </div>
+    <Loading v-if="loading" />
+  </div>
+</template>
+
+<script>
+import { debounce } from 'lodash-es';
+import Loading from '../../Loading.vue';
+import IconSearch from '../../../assets/iconSearch.svg?icon-component';
+
+export default {
+  components: {
+    IconSearch,
+    Loading,
+  },
+  data() {
+    return {
+      query: '',
+      results: [],
+      offset: 0,
+      resultsCount: '',
+      loading: false,
+    };
+  },
+  mounted() {
+    this.search();
+  },
+  methods: {
+    submit: debounce(function a() {
+      this.resultsCount = '';
+      this.loading = true;
+      this.results = [];
+      this.offset = 0;
+      this.search();
+    }, 500),
+    search() {
+      return fetch(`https://api.giphy.com/v1/gifs/${this.query ? `search?q=${this.query}&` : 'trending?'}api_key=${process.env.VUE_APP_GIPHY_API_KEY}&limit=10&offset=${this.offset}`)
+        .then((res) => res.json())
+        .then(({ data, pagination }) => {
+          this.loading = false;
+          this.resultsCount = `${pagination.total_count.toLocaleString()} Results`;
+          this.results.push(...data.map(({ images }) => ({
+            still: images.fixed_width_still.url,
+            animated: images.fixed_width.url,
+            original: images.original.url,
+          })));
+        })
+        .catch((e) => console.log(e));
+    },
+    loadMore() {
+      const { clientHeight, scrollTop, scrollHeight } = this.$refs.gifsContainer;
+      if (clientHeight + scrollTop >= scrollHeight - 1 && this.loading === false) {
+        this.loading = true;
+        this.offset += 10;
+        this.search();
+      }
+    },
+    toggleStill(element, animated) {
+      element.setAttribute('src', animated);
```

Comment:
Let vue to manage the DOM nodes by itself, I'm proposing to remember the index here and to switch the image by it.